src/panels/config/integrations/ha-config-integrations.ts

```diff
@@ -185,22 +218,30 @@ class HaConfigIntegrations extends SubscribeMixin(LitElement) {
       this._configEntries.length
     ) {
       afterNextRender(() => {
-        const card = this.shadowRoot!.getElementById(
-          this._searchParms.get("config_entry")!
+        const entryId = this._searchParms.get("config_entry")!;
+        const configEntry = this._configEntries.find(
+          (entry) => entry.entry_id === entryId
         );
+        if (!configEntry) {
+          return;
+        }
+        const card: HaIntegrationCard = this.shadowRoot!.getElementById(
+          configEntry?.domain
+        ) as HaIntegrationCard;
```

Comment:
This can break cast SDK 😆 Let's not do it. Use `this.shadowRoot.querySelector(\`[data-domain=${domain}]\`)`

See https://github.com/home-assistant/frontend/blob/dev/src/cast/cast_framework.ts#L13-L15